File: app/src/main/java/org/vosk/service/utils/Tools.java
package org.vosk.service.utils;

import android.app.ActivityManager;
import android.content.Context;

import org.jetbrains.annotations.NotNull;
import org.vosk.service.download.DownloadModelService;

import java.io.File;

public class Tools {

    public static File getModelFileRootPath(@NotNull Context context) {
        return new File(context.getFilesDir(), "models");
    }

    public static boolean isServiceRunning(Context context) {
        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
            if (DownloadModelService.class.getName().equals(service.service.getClassName())) {
                return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/org/vosk/service/utils/Tools.kt
package org.vosk.service.utils

import android.app.ActivityManager
import android.content.Context
import org.vosk.service.download.DownloadModelService
import java.io.File

object Tools {
	@JvmStatic
    fun getModelFileRootPath(context: Context): File {
		return File(context.filesDir, "models")
	}

	@JvmStatic
    fun isServiceRunning(context: Context): Boolean {
		val manager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
		for (service in manager.getRunningServices(Int.MAX_VALUE)) {
			if (DownloadModelService::class.java.name == service.service.className) {
				return true
			}
		}
		return false
	}
}